src/lib/update-lock.ts

```diff
@@ -87,6 +87,29 @@ function dispose(
 		.return();
 }
 
+/**
+ * Attempts to take locks for list of application IDs provided
+ * The locks will exist until all applications are locked and the passed function resolves or throws.
+ */
+export async function lockAll<T extends unknown>(
+	appIds: number[],
+	force = false,
+	fn: () => Resolvable<T>,
+): Promise<T | undefined> {
+	if (appIds.length === 0) {
+		return;
+	}
+	let lockCounter = 0;
+	const nextLock = async (): Promise<Resolvable<T>> => {
+		lockCounter++;
+		if (lockCounter === appIds.length) {
+			return lock(appIds[lockCounter + 1], { force }, fn);
+		}
+		return lock(appIds[lockCounter + 1], { force }, nextLock);
+	};
+	return lock(appIds[0], { force }, nextLock);
```

Comment:
If there is 1 application the lock is called twice which will result in the 2nd call throwing since locks still exist from the first one.

Reply:
resolved by calling fn if there is only 1 appId in the array